service/history/timer_queue_active_task_executor.go

```diff
@@ -925,6 +929,62 @@ func (t *timerQueueActiveTaskExecutor) processActivityWorkflowRules(
 	return nil
 }
 
+func (t *timerQueueActiveTaskExecutor) executeChasmSideEffectTimerTask(
+	ctx context.Context,
+	task *tasks.ChasmTask,
+) error {
+	ctx, cancel := context.WithTimeout(ctx, taskTimeout)
+	defer cancel()
+
+	if !queues.IsTimeExpired(task, t.Now(), task.GetVisibilityTime()) {
+		return nil
+	}
+
+	wfCtx, release, err := getWorkflowExecutionContextForTask(ctx, t.shardContext, t.cache, task)
+	if err != nil {
+		return err
+	}
+	defer func() { release(err) }()
+
+	ms, err := loadMutableStateForTimerTask(ctx, t.shardContext, wfCtx, task, t.metricsHandler, t.logger)
+	if err != nil {
+		return err
+	}
+	if ms == nil {
+		return nil
+	}
```

Comment:
also should never happen once the eventID check is fixed (bypassed essentially).